op-challenger/game/service.go

```diff
@@ -79,24 +102,20 @@ func NewService(ctx context.Context, logger log.Logger, cfg *config.Config) (*Se
 
 	pollClient, err := opClient.NewRPCWithClient(ctx, logger, cfg.L1EthRpc, opClient.NewBaseRPCClient(l1Client.Client()), cfg.PollInterval)
 	if err != nil {
-		return nil, fmt.Errorf("failed to create RPC client: %w", err)
+		return nil, errors.Join(fmt.Errorf("failed to create RPC client: %w", err), s.Close())
 	}
-	monitor := newGameMonitor(logger, cl, loader, sched, cfg.GameWindow, l1Client.BlockNumber, cfg.GameAllowlist, pollClient)
+	s.monitor = newGameMonitor(logger, cl, loader, s.sched, cfg.GameWindow, l1Client.BlockNumber, cfg.GameAllowlist, pollClient)
 
 	m.RecordInfo(version.SimpleWithMeta)
 	m.RecordUp()
 
-	return &Service{
-		logger:  logger,
-		metrics: m,
-		monitor: monitor,
-		sched:   sched,
-	}, nil
+	return s, nil
 }
 
 // MonitorGame monitors the fault dispute game and attempts to progress it.
 func (s *Service) MonitorGame(ctx context.Context) error {
 	s.sched.Start(ctx)
 	defer s.sched.Close()
+	defer s.Close()
```

Comment:
Since the `Service` `s` owns its `sched`, I suggest to move `s.sched.Close()` into `Service.Close()`.